python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1510,6 +1506,26 @@ def get_run():
         except DagsterRunAlreadyExists:
             return get_run()
 
+    def handle_started_run(self, run_record: RunRecord, logger: logging.Logger) -> None:
+        max_time_str = run_record.dagster_run.tags.get(MAX_RUNTIME_TAG)
+
+        # If MAX_RUNTIME_TAG has not been set, nothing to do
+        if not max_time_str:
+            return
```

Comment:
falling back to an optional system-wide default seems good here? So you can just say "kill any run that goes over 12 hours that should never happen"